### PR TITLE
implement the low cardinality dictionary

### PR DESCRIPTION
**Pull request by qqIsAProgrammer**
implement the low cardinality dictionary

## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/container/dict/dict.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
+package dict
```
Comment (aunjgr) [inline pkg/container/dict/dict.go:1]: Please prepend apache 2.0 license header to every .go source file.

You can refer to any existing *.go

**Review by aunjgr — CHANGES_REQUESTED** [review]

**Review by nnsgmsone — CHANGES_REQUESTED** [review]
Please add bvt and performance comparison

File: pkg/container/dict/dict.go (hunk 1 above)
Comment (qqIsAProgrammer) [reply pkg/container/dict/dict.go:15]: Okay, done.

**Comment by qqIsAProgrammer** [comment]
> Please add bvt and performance comparison

Currently only the data structure has been implemented, and it is not able to add BVT. More tests will be added when the feature is completed.

**Review by aunjgr — APPROVED** [review]

**Comment by nnsgmsone** [comment]
> > Please add bvt and performance comparison
> 
> Currently only the data structure has been implemented, and it is unable to add BVT. More tests will be added when the feature is completed.


Can you add a comparison test with the current hashtable's benchmark?

**Comment by qqIsAProgrammer** [comment]
> > > Please add bvt and performance comparison
> > 
> > 
> > Currently only the data structure has been implemented, and it is unable to add BVT. More tests will be added when the feature is completed.
> 
> Can you add a comparison test with the current hashtable's benchmark?

I'm sorry I didn't make that clear. The Dict is not a hashtable, but a data structure used in LowCardinality type. Currently it's unable to do performance tests, but will add them when LowCardinality type is completed.

**Pull request closed without merging**